Formula/nauty.rb

```diff
@@ -1,13 +1,14 @@
 class Nauty < Formula
   desc "Automorphism groups of graphs and digraphs"
   homepage "https://pallini.di.uniroma1.it/"
-  url "https://pallini.di.uniroma1.it/nauty27r1.tar.gz"
-  version "27r1"
-  sha256 "76ca5d196e402c83a987f90c28ff706bcc5a333bb4a8fbb979a62d3b99c34e77"
+  url "https://pallini.di.uniroma1.it/nauty27r2.tar.gz"
+  version "2.7r2"
+  sha256 "69c16c17da1cca90c25f6fb3bb242798af95f096fc3e2ff3e3398f390fcea768"
+  version_scheme 1
 
   livecheck do
     url :homepage
-    regex(/href=.*?nauty[._-]?v?(\d+(?:\.\d+)*(?:r\d+)?)\.t/i)
+    regex(/Current\s+version:\s+(\d+\.?\d+r\d+)/i)
```

Comment:
```suggestion
    regex(/Current\s+?version:\s*?v?(\d+(?:\.\d+)+(?:r\d+)?)/i)
```

If we go the simple route and match the formatted version from the "Current version: ..." text, we should make the dot between the major and minor version required (like the standard regex for versions like `1.2.3`/`v1.2.3`, `v?(\d+(?:\.\d+)+)`). If this text ever omits dots in the version (e.g., `Current version: 27r2`), then we wouldn't be any better off taking this approach over matching the version from the tarball file name and naively manipulating it in a `strategy` block.

This suggestion also makes the trailing `r\d+` part optional, as in the existing regex. Some of the older versions (e.g., `2.2`) don't include a trailing `r\d+` part, so we want this to continue to match if we encounter a version like this in the future.

In general, we try to avoid loose checks like this but the alternative option (naively adding dots in a `strategy` block) isn't great either, so I'm good with this approach for now. We can always revisit this if the situation changes in the future (for better or worse).